casper/src/main/resources/PoS.rhox

```diff
@@ -268,37 +292,39 @@ new  PoS,
                   }
                 } |
 
-                contract removeQuarantinedWithdrawers(@currentBlockNumber, @allBonds, @withdrawers, newWithdrawersCh, newBondsCh) = {
+                contract removeQuarantinedWithdrawers(@currentBlockNumber, @allBonds, @withdrawers, @rewards, returnCh) = {
                   new quarantinedValidatorsCh,
                       validatorsToWithdrawListCh,
                       isQuarantineFinished,
-                      isWithdrawn,
+                      notWithdrawn,
                       newBondsListCh,
-                      newWithdrawersListCh
+                      newWithdrawersListCh,
+                      newRewardsListCh,
+                      paymentsCh, computePay
                   in {
                     @ListOps!("filter", withdrawers.toList(), *isQuarantineFinished, *quarantinedValidatorsCh) |
                     for (@quarantinedValidators <- quarantinedValidatorsCh) {
                       @ListOps!("map", quarantinedValidators, *fst, *validatorsToWithdrawListCh) |
                       for (@validatorsToWithdrawList <- validatorsToWithdrawListCh) {
+                        @ListOps!("map", validatorsToWithdrawList, *computePay, *paymentsCh) |
+                        contract computePay(@pk, resultCh) = {
+                          resultCh!((pk, allBonds.get(pk) + rewards.getOrElse(pk, 0)))
```

Comment:
Is `withDraw` basically un-bonding?

Reply:
it is. I'm not sure why the name changed. Maybe @KentShikama knows